File: PB - More Exercises Python/4.0 While-Loop - More Exercises/03. Stream Of Letters.py
word, collection, code = '', '', ['c', 'n', 'o']

for ch in [x for x in iter(input, 'End') if x.isalpha()]:
    if ch in code:
        code.remove(ch)
    else:
        word += ch
    if not len(code):
        code = ['c', 'n', 'o']
        collection += word
        word = ' '

print(collection)


# --------------------------------------------------------(2)-----------------------------------------

chars = input()
word_collection = ''
word = ''
c, o, n = False, False, False

while chars != 'End':
    if chars.isalpha():
       if (chars =='c' and c != True) or (chars =='o' and o != True) or (chars =='n' and n != True):
            if chars =='c': c = True
            if chars =='o': o = True
            if chars =='n': n = True
       else:
          word += chars

    if c and o and n:
       word_collection += word + ' '
       c, o, n = False, False, False
       word = ''

    chars =  input()

print(word_collection)


'''
H
n
e
l
l
o
o
c
t
c
h
o
e
r
e
n
e
End
'''

File: PB - More Exercises GO/4.0 While-Loop - More Exercises/03. Stream Of Letters.go
package main

import "fmt"
import  "regexp"
func main() {
   var wordCollection, word string
   var c, o, n bool
   
   for {
      var chars string
      fmt.Scan(&chars)
      if chars == "End"{
         break
      }

      var digitCheck = regexp.MustCompile(`[^A-Za-z]+$`)

      if (digitCheck.MatchString(chars)) {
         continue
      }


      if (chars == "c" && c != true) || (chars == "o" && o != true) || (chars == "n" && n != true) {

         if chars == "c"{
             c = true
         }
         if chars == "o"{
             o = true
         }
         if chars == "n"{
            n = true
         } 
      } else {
         word += chars
      }

      if (c && o && n) {
         wordCollection += word + " "
         c = false
         o = false
         n = false
         word = ""
      }
   }

   fmt.Println(wordCollection)
}

/*
name   :03. Stream Of Letters
input  :H n e l l o o c t c h o e r e n e End
output :Hello there 
*/
